packages/j/joltphysics/xmake.lua

```diff
@@ -74,9 +76,75 @@ package("joltphysics")
+            -- add instruction sets (from https://github.com/jrouwe/JoltPhysics/blob/4cd52055e09160affcafa557b39520331bf0d034/Jolt/Jolt.cmake#L602)
+            if package:has_tool("cxx", "cl") then
+                if package:config("inst_avx512") then
+                    package:add("cxxflags", "/arch:AVX512")
+                elseif package:config("inst_avx2") then
+                    package:add("cxxflags", "/arch:AVX2")
+                elseif package:config("inst_avx") then
+                    package:add("cxxflags", "/arch:AVX2")
+                end
+            else
+                if package:config("inst_avx512") then
+                    package:add("cxxflags", "-mavx512f", "-mavx512vl", "-mavx512dq", "-mavx2", "-mbmi", "-mpopcnt", "-mlzcnt", "-mf16c")
+                elseif package:config("inst_avx2") then
+                    package:add("cxxflags", "-mavx2", "-mbmi", "-mpopcnt", "-mlzcnt", "-mf16c")
+                elseif package:config("inst_avx") then
+                    package:add("cxxflags", "-mavx", "-mpopcnt")
+                elseif package:config("inst_sse4_2") then
+                    package:add("cxxflags", "-msse4.2", "-mpopcnt")
+                elseif package:config("inst_sse4_1") then
+                    package:add("cxxflags", "-msse4.1")
+                else
+                    package:add("cxxflags", "-msse2")
+                end
+                if package:config("inst_lzcnt") then
+                    package:add("cxxflags", "-mlzcnt")
+                end
+                if package:config("inst_tzcnt") then
+                    package:add("cxxflags", "-mbmi")
+                end
+                if package:config("inst_f16c") then
+                    package:add("cxxflags", "-mf16c")
+                end
+                if package:config("inst_fmadd") and not package:config("cross_platform_deterministic") then
+                    package:add("cxxflags", "-mfma")
+                end
+                package:add("cxxflags", "-mfpmath=sse")
+            end
+            if package:config("inst_avx512") then
+        		package:add("defines", "JPH_USE_AVX512")
+            end
+            if package:config("inst_avx2") then
+        		package:add("defines", "JPH_USE_AVX2")
+            end
+            if package:config("inst_avx") then
+        		package:add("defines", "JPH_USE_AVX")
+            end
+        	if package:config("inst_sse4_1") then
+        		package:add("defines", "JPH_USE_SSE4_1")
+            end
+        	if package:config("inst_sse4_2") then
+        		package:add("defines", "JPH_USE_SSE4_2")
+            end
+        	if package:config("inst_lzcnt") then
+        		package:add("defines", "JPH_USE_LZCNT")
+            end
+        	if package:config("inst_tzcnt") then
+        		package:add("defines", "JPH_USE_TZCNT")
+            end
+        	if package:config("inst_f16c") then
+        		package:add("defines", "JPH_USE_F16C")
+            end
```

Comment:
Try?
```lua
local configs = {"inst_avx512"}
for _, config in ipairs(configs) do
    if package:config(config) then
        package:add("defines", "JPH_USE_" .. string.match(config, "^inst_(.+)$"):upper())
    end
end
```

Reply:
I just translated the Jolt.cmake, I think it'll be easier for package updates